server/src/main/java/org/elasticsearch/cluster/coordination/StableMasterHealthIndicatorService.java

```diff
@@ -0,0 +1,449 @@
+     * and "recent_masters"
+     */
+    private HealthIndicatorDetails getSimpleDetails(boolean explain, MasterHistory localMasterHistory) {
+        return explain ? (builder, params) -> {
+            builder.startObject();
+            DiscoveryNode masterNode = localMasterHistory.getMostRecentMaster();
+            builder.object(DETAILS_CURRENT_MASTER, xContentBuilder -> {
+                if (masterNode != null) {
+                    builder.field("node_id", masterNode.getId());
+                    builder.field("name", masterNode.getName());
+                } else {
+                    builder.nullField("node_id");
+                    builder.nullField("name");
+                }
+            });
+            List<DiscoveryNode> recentMasters = localMasterHistory.getNodes();
+            builder.array(DETAILS_RECENT_MASTERS, arrayXContentBuilder -> {
+                for (DiscoveryNode recentMaster : recentMasters) {
+                    if (recentMaster != null) {
+                        builder.startObject();
+                        builder.field("node_id", recentMaster.getId());
+                        builder.field("name", recentMaster.getName());
+                        builder.endObject();
+                    }
+                }
+            });
+            return builder.endObject();
+        } : HealthIndicatorDetails.EMPTY;
+    }
+
+    /**
+     * This method returns the only user action that is relevant when the master is unstable -- contact support.
+     * @param explain If true, the returned list includes a UserAction to contact support, otherwise an empty list
+     * @return a single UserAction instructing users to contact support.
+     */
+    private List<UserAction> getContactSupportUserActions(boolean explain) {
+        if (explain) {
+            UserAction.Definition contactSupport = new UserAction.Definition(
+                "contact_support",
+                "The Elasticsearch cluster does not have a stable master node. Please contact Elastic Support "
+                    + "(https://support.elastic.co) to discuss available options.",
+                null
+            );
+            UserAction userAction = new UserAction(contactSupport, null);
+            return List.of(userAction);
+        } else {
+            return List.of();
+        }
+    }
+
+    /**
+     * Returns the list of the impacts of an unstable master node.
+     * @return The list of the impacts of an unstable master node
+     */
+    private List<HealthIndicatorImpact> getUnstableMasterImpacts() {
+        List<HealthIndicatorImpact> impacts = new ArrayList<>();
+        impacts.add(new HealthIndicatorImpact(1, UNSTABLE_MASTER_INGEST_IMPACT, List.of(ImpactArea.INGEST)));
+        impacts.add(new HealthIndicatorImpact(1, UNSTABLE_MASTER_DEPLOYMENT_MANAGEMENT_IMPACT, List.of(ImpactArea.DEPLOYMENT_MANAGEMENT)));
+        impacts.add(new HealthIndicatorImpact(3, UNSTABLE_MASTER_BACKUP_IMPACT, List.of(ImpactArea.BACKUP)));
```

Comment:
We might as well hard-code this as a `static` list instead of constructing it every time, right? Since it doesn't change based on anything else.